Rubberduck.Core/UI/Command/RunAllTestsCommand.cs

```diff
@@ -18,104 +11,26 @@ public class RunAllTestsCommand : CommandBase
     {
         private readonly IVBE _vbe;
         private readonly ITestEngine _engine;
-        private readonly TestExplorerModel _model;
-        private readonly IDockablePresenter _presenter;
-        private readonly RubberduckParserState _state;
 
-        public RunAllTestsCommand(IVBE vbe, RubberduckParserState state, ITestEngine engine, TestExplorerModel model, IDockablePresenter presenter) 
+        public RunAllTestsCommand(IVBE vbe, ITestEngine engine)
             : base(LogManager.GetCurrentClassLogger())
         {
             _vbe = vbe;
             _engine = engine;
-            _model = model;
-            _state = state;
-            _presenter = presenter;
         }
 
         protected override bool EvaluateCanExecute(object parameter)
         {
-            return _vbe.IsInDesignMode && _engine.AllowedRunStates.Contains(_state.Status);
+            // the vbe design mode requirement could also be encapsulated into the engine
```

Comment:
Is it just me or checking `_vbe.IsInDesignMode` is the only reason we're taking an `IVBE` dependency here? Unless I missed another use somewhere, moving the design mode check to the `ITestEngine` implementation seems like a very good idea!

Reply:
The point of deliberation here is that the TestEngine does not currently take an IVBE as dependency.

Reply:
IMO that's a non-issue - taking in dependencies *of dependencies* is though: TestEngine looks like it *wants* to be able to determine whether the VBE is in design mode.. that would be *feature envy*, and satisfying it would move the `IVBE` dependency where it wants to be. No?